Update LSP12 Issued Asset Guides (#943)
lukso-network/docs: docs/learn/dapp-developer/register-lsp12-issued-assets.md

---
sidebar_label: '🪪 Register Issued Assets'
sidebar_position: 13
description: Learn how to register Issued Assets under an Universal Profile.
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

# Register Issued Assets

:::success Asset Authenticity

> During the [creation of an asset](./deploy-contracts.md), Universal Profiles can be referenced as creators within the [LSP4 Metadata](../../standards/tokens/LSP4-Digital-Asset-Metadata.md). To **authenticate the creators** of an asset, the corresponding Universal Profiles are required to also reference the asset using [LSP12 Issued Assets](../../standards/universal-profile/lsp12-issued-assets.md). Services can then validate the **on-chain cross-link** to prove authenticity.

Reviewer: ```suggestion
During the [creation of an asset](./deploy-contracts.md), Universal Profiles can be referenced as creators within the [LSP4 Metadata](../../standards/tokens/LSP4-Digital-Asset-Metadata.md). To **authenticate the creators** of an asset, their Universal Profiles must list the asset's address under the [LSP12 Issued Assets](../../standards/universal-profile/lsp12-issued-assets.md). Services can then validate the **on-chain cross-link** to prove authenticity.
```

  //   },
];

const web3 = new Web3(window.lukso);
await web3.eth.requestAccounts();
const accounts = await web3.eth.getAccounts();
```

  </TabItem>
</Tabs>

## Encode the data keys

> After setting up the array of assets, you can use the [`erc725.js`](../../tools/erc725js/getting-started.md) library and the [LSP12 JSON Schema](../../standards/universal-profile/lsp12-issued-assets/) to encode the data keys. There are two ways to encode data:

Reviewer: ```suggestion
After setting up the array of assets, you can use the [`erc725.js`](../../tools/erc725js/getting-started.md) library to encode the [`LSP12IssuedAssets[]`](../../standards/universal-profile/lsp12-issued-assets/) data keys. There are two ways to encode data:
```

- **Set the full issued assets list**: If you want to initially set issued assets on a Universal Profile or re-set all elements, you can encode the data without defining optional length or index parameters. Therefore, the issued assets will only consist of the asset addresses you provide in your array of assets.

> - **add, update, or remove existing issued assets**: If the Universal Profile already has issued assets, and you want to add, update, or remove certain assets, you can provide the `startingIndex` and `totalArrayLength` parameters. Therefore, your prepared array of assets only represents a subset of the total issued assets set on the contract.

Reviewer: ```suggestion
- **add, update, or remove existing issued assets**: If the Universal Profile already has issued assets, and you want to add, update, or remove certain assets, you can provide the `startingIndex` and `totalArrayLength` parameters. Therefore, your prepared array only represents a subset of all the issued assets listed in your Universal Profile.
```

  UniversalProfileArtifact.abi,
  myWallet,
);
```

  </TabItem>
    <TabItem value="web3js" label="web3.js">

```javascript
const myUPContract = new web3.eth.Contract(
  UniversalProfileArtifact.abi,
  UNIVERSAL_PROFILE_ADDRESS,
);
```

  </TabItem>
</Tabs>

## Set data batch

Next, use the `setDataBatch(...)` function of the Universal Profile to initially set or update multiple data keys.

<Tabs groupId="web3-lib">

>     <TabItem value="ethersjs" label="ethers.js">

Reviewer: ```suggestion
    <TabItem value="ethersjs" label="ethers.js" default>
```
